client/jetpack-connect/authorize.js

```diff
@@ -632,15 +632,15 @@ export class JetpackAuthorize extends Component {
 
 export default connect(
 	( state, { authQuery } ) => {
-		const siteSlug = urlToSlug( authQuery.site );
+		const siteSlug = urlToSlug( authQuery.homeUrl );
```

Comment:
If we're not using this for other connected props, this is _only_ used for redirection now. Let's just do this in the redirection where it's used.

I'll push a change.